db/disk/submission.go

```diff
@@ -294,3 +294,30 @@ func (this *backend) getMostRecentSubmissionID(assignment *model.Assignment, ema
 
     return dirents[len(dirents) - 1].Name(), nil;
 }
+
+func (this *backend) RemoveSubmission(assignment *model.Assignment, email string, shortSubmissionID string) (bool, error) {
+    var err error;
+
+    if (shortSubmissionID == "") {
+        shortSubmissionID, err = this.getMostRecentSubmissionID(assignment, email);
+        if (err != nil){
+            return false, fmt.Errorf("Failed to get most recent submission id: `%w`.", err);
+        }
+    }
+
+    if (shortSubmissionID == ""){
+        return false, nil;
+    }
+
+    submissionDir := this.getSubmissionDirFromAssignment(assignment, email, shortSubmissionID);
+
+    if (!util.PathExists(submissionDir)) {
+        return false, fmt.Errorf("Submission does not exist.");
```

Comment:
This is not an error condition.
The whole reason to have the first return value (the boolean) is so that we can tell when a submission actually existed.
So there is no reason to also create an error in this case.

Also, think about this from the "recent submission" case (where the submission ID is empty).
It seems mean to punish someone with an error for removing an ID that they didn't pass in.
(They could have tried to get the submission first, but it still seems like we are putting a big onus on the caller.)